features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIProduct.jsx

```diff
@@ -370,6 +385,47 @@ class APIProduct extends Resource {
 
         return promiseAddFileToDocument;
     }
+
+    /**
+     * Update an api swagger via PUT HTTP method
+     * @param {*} swagger 
+     */
+    updateSwagger(swagger) {
+        const promisedUpdate = this.client.then((client) => {
+            const payload = {
+                apiProductId: this.id,
+                apiDefinition: JSON.stringify(swagger),
+                'Content-Type': 'multipart/form-data',
+            };
+            return client.apis['API Product (Individual)'].put_api_products__apiProductId__swagger(
+                payload,
+                this._requestMetaData({
+                    'Content-Type': 'multipart/form-data'
+                }),
+            ).catch((error) => {
+                console.error(error);
+            });
+        });
+        return promisedUpdate.then(response => {
+            return this;
+        });
+    }
+
+    /**
+     * Get the swagger of an API Product
+     * @param id {String} UUID of the API Product in which the swagger is needed
+
```

Comment:
shall we remove this extra space

Reply:
fixed with https://github.com/wso2/carbon-apimgt/commit/a5b4d09fd25478c04985b9e5f2eea62f56fed44d